example/src/App.tsx

```diff
@@ -364,6 +382,64 @@ const App = () => {
             setIsProgrammaticMRecShowing(!isProgrammaticMRecShowing);
           }}
         />
+        <View style={{paddingHorizontal: 35}}>
+          <View style={{flexDirection: 'row', justifyContent: 'space-between'}}>
+            <AppButton
+              title={'Preload Native Ad'}
+              enabled={AppLovinMAX.isInitialized()}
+              onPress={() => {
+                AppLovinMAX.preLoadNativeAd(NATIVE_AD_UNIT_ID);
+              }}
+              style={{paddingLeft: 5, paddingRight: 2}}
+            />
+            <AppButton
+              title={isNativeAdShowing ? 'Hide' : 'Show'}
+              enabled={AppLovinMAX.isInitialized()}
+              onPress={() => {
+                setIsNativeAdShowing(!isNativeAdShowing);
+              }}
+              style={{paddingLeft: 2, paddingRight: 2}}
+            />
+            <AppButton
+              title={'Load'}
+              enabled={AppLovinMAX.isInitialized()}
+              onPress={() => {
+                nativeAdViewRef.current?.loadAd();
+               }}
+              style={{paddingLeft: 2, paddingRight: 5}}
+            />
+          </View>
+        </View>
+        {
+          isNativeAdShowing &&
+            <AppLovinMAX.NativeAdView adUnitId={NATIVE_AD_UNIT_ID}
+                                      placement='myplacement'
+                                      customData='mycustomdata'
+                                      extraParameter={{
+                                        'key1': 'value1',
+                                        'key2': 'value2',
+                                      }}
+                                      ref={nativeAdViewRef}
+                                      style={styles.nativead}>
+              <View style={{flex: 1, flexDirection: 'column'}}>
+                <View style={{flexDirection: 'row'}}>
+                  <AppLovinMAX.IconView style={styles.icon}/>
+                  <View style={{flexDirection: 'column', flexGrow: 1}}>
+                    <AppLovinMAX.TitleView style={styles.title}/>
+                    <View style={{flexDirection: 'row',
+                                  justifyContent: 'space-between',
+                                  paddingHorizontal: 10}}>
+                      <AppLovinMAX.AdvertiserView style={styles.advertiser}/>
+                      <Text style={styles.admark}>AD</Text>
+                    </View>
+                  </View>
+                </View>
+                <AppLovinMAX.BodyView style={styles.body}/>
+                <AppLovinMAX.MediaView style={styles.mediaView}/>
+                <AppLovinMAX.CallToActionView style={styles.callToAction}/>
+              </View>
+            </AppLovinMAX.NativeAdView>
```

Comment:
Can we just break this whole component and its styling out into its own `NativeAdView`?

Reply:
Externalized to NativeAdView.